snapshot/src/main/java/io/zeebe/snapshots/broker/SnapshotStoreSupplier.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.snapshots.broker;
+
+import io.zeebe.snapshots.raft.PersistedSnapshotStore;
+import io.zeebe.snapshots.raft.ReceivableSnapshotStore;
+
+public interface SnapshotStoreSupplier {
+
+  /**
+   * Returns a partition's {@link ConstructableSnapshotStore}
+   *
+   * @param partitionName
+   * @return a ConstructableSnapshotStore
+   */
+  ConstructableSnapshotStore getConstructableSnapshotStore(String partitionName);
+
+  /**
+   * Returns a partition's {@link ReceivableSnapshotStore}
+   *
+   * @param partitionName
+   * @return a ReceivableSnapshotStore
+   */
+  ReceivableSnapshotStore getReceivableSnapshotStore(final String partitionName);
+
+  /**
+   * Returns a partition's {@link PersistedSnapshotStore}
+   *
+   * @param partitionName
+   * @return a ConstructableSnapshotStore
+   */
+  PersistedSnapshotStore getPersistedSnapshotStore(String partitionName);
```

Comment:
Do we need that?

Reply:
LogDeletionService expects only PersistedSnapshotStore. It doesn't care if it is Receivable or Constructable.

Reply:
But the other interfaces extend this one right? So you can just give one of them? Not sure whether it makes sense to have this getter

Reply:
Then the question is which one to use. Both would pass the compiler. But conceptually, on the leader it does not make sense  to use `RecievableStore` and on the follower to use `ConstructableStore`.

Reply:
sounds not ideal but ok should be fine for now

Reply:
Thanks. I don't like it either. But could not find a better way. :)  Any other comments?